.github/workflows/ci-shortfin-developer-tool-tests.yml

```diff
@@ -0,0 +1,87 @@
+# Copyright 2024 Advanced Micro Devices, Inc.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+name: Shortfin Developer Tools Tests
+
+on:
+  workflow_dispatch:
+  schedule:
+    # Run weekly to ensure our tools remain functional
+    - cron: '0 0 * * 0'
+  pull_request:
+    paths:
+      - '.github/workflows/ci-shortfin-developer-tool-tests.yml'
+      - 'app_tests/integration_tests/llm/shortfin/tracing_test.py'
+      - 'shortfin/**'
+
+concurrency:
+  # A PR number if a pull request and otherwise the commit hash. This cancels
+  # queued and in-progress runs for the same PR (presubmit) or commit
+  # (postsubmit). The workflow name is prepended to avoid conflicts between
+  # different workflows.
+  group: ${{ github.workflow }}-${{ github.event.number || github.sha }}
+  cancel-in-progress: true
+
+jobs:
+  shortfin_tracing_test:
+    name: "Shortfin Tracing Test"
+    strategy:
+      matrix:
+        version: [3.11]
+      fail-fast: false
+    runs-on: linux-mi300-1gpu-ossci
```

Comment:
Sanity checking a developer tool should be possible on a github-hosted CPU runner, no GPU needed.